lib/ansible/module_utils/facts/timeout.py

```diff
@@ -25,7 +25,7 @@
 DEFAULT_GATHER_TIMEOUT = 10
 
 
-class TimeoutError(Exception):
+class TimeoutError(BaseException):
```

Comment:
BaseException should not be used for application exceptions.

 exception BaseException

https://docs.python.org/2/library/exceptions.html#exceptions.BaseException
   

>  The base class for all built-in exceptions. It is not meant to be directly inherited by user-defined classes (for that, use Exception). 

https://docs.python.org/2/library/exceptions.html#exceptions.Exception

> All built-in, non-system-exiting exceptions are derived from this class. All user-defined exceptions should also be derived from this class.